Tools/TestWebKitAPI/Tests/IPC/IPCSerialization.mm

```diff
@@ -530,6 +531,21 @@ static bool wkNSURLProtectionSpace_isEqual(NSURLProtectionSpace *a, SEL, NSURLPr
     return true;
 }
 
+static bool NSURLCredentialTesting_isEqual(NSURLCredential *a, NSURLCredential *b)
+{
+    if (a.persistence != b.persistence)
+        return false;
+    if (a.user && ![a.user isEqualToString:b.user])
+        return false;
+    if (a.password && ![a.password isEqualToString:b.password])
+        return false;
+    if (a.hasPassword != b.hasPassword)
+        return false;
+    if (a.certificates.count != b.certificates.count)
```

Comment:
we should probably make sure the 
`@property (nullable, readonly) SecIdentityRef identity;`

matches.

Reply:
also SecTrustRef

Reply:
We don't have a way to compare SecIdentityRefs right now, but this functionality is tested in the test TestWebKitAPI.Challenge.ClientCertificate.

NSURLCredential has no accessor for a SecTrustRef, but NSURLCredential.credentialForTrust is also well tested.

Reply:
I filed rdar://123342292